pkg/build/builder/cmd/builder.go

```diff
@@ -50,6 +50,7 @@ func run(builderFactory factoryFunc) {
 				glog.Fatalf("Cannot setup secret file for accessing private repository: %v", err)
 			}
 			if sourceURL != nil {
+				build.Annotations[bld.OriginalSourceURLAnnotationKey] = build.Spec.Source.Git.URI
```

Comment:
nm, i see this is only on the "internal" build object in the build container, so it's not exposed into the api.